### ISSUE TITLE
Generate code coverage for javascript tests

### ISSUE THREAD

**Issue by sungwoncho** [dnote/dnote#266]
* **Please describe the feature you would like to have.**
When running `web` unit tests using Karma, generate a code coverage.

```
make test-web
```

* **What alternatives have you considered?**
None

* **Any additional context**

* Idea: use [istanbul](https://istanbul.js.org/)

**Comment by geritol** [comment]
working on it!

**Comment by geritol** [comment]
Created a command `npm run test:coverage`. Should it be ran instead of `test` when running `make test-web` without watch mode?

**Comment by sungwoncho** [comment]
@geritol thanks for the PR. `It looks like the coverage is not properly generated. For instance, even though `web/src/libs/editor.ts` is tested, the output reveals:

```
--------------------------------------------------------|----------|----------|----------|----------|-------------------|
File                                                    |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
--------------------------------------------------------|----------|----------|----------|----------|-------------------|
...
  editor.ts                                             |        0 |        0 |        0 |        0 |       20,26,27,30 |
```

Upon digging further, I found that `nyc` might not be the right tool since the tests are being run in a browser environment by Karma, rather than in a node.js process. (https://github.com/istanbuljs/nyc/issues/203#issuecomment-201118963)

I will close this for now, and I will take a crack at it later.

**Comment by sungwoncho** [comment]
Implemented by #302. Thanks for your input.
